tests/integration/fixtures/docker_test_image/test.py

```diff
@@ -90,5 +90,20 @@ def listdir(d):
         print(path)
 
 
+@cli.command()
+@click.argument('f1', type=click.File('r'))
+@click.argument('f2', type=click.File('r'))
+@click.argument('out', type=click.File('w'))
+def concat(f1, f2, out):
+    out.write(f1.read() + f2.read())
+
+
+@cli.command()
+@click.argument('f', type=click.File('r'))
+@click.argument('out', type=click.File('w'))
+def reverse(f, out):
+    out.write(reversed(f.read()))
```

Comment:
Does this line work because of some difference with `click.File('w')`?

```python
In[2]: with open('hmm.txt', 'w') as f:
  ...:     f.write('123\n456\n789\n')
  ...: 
In[3]: with open('hmm.txt', 'r') as f:
  ...:     content = f.read()
  ...: 
In[4]: with open('out.txt', 'w') as f:
  ...:     f.write(reversed(content))
  ...: 
Traceback (most recent call last):
  File "", line 3265, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-4-844f453fde68>", line 2, in <module>
    f.write(reversed(content))
TypeError: write() argument must be str, not reversed
```

Maybe `out.write(f.read()[::-1])` ?

Reply:
You're right, I hadn't tested this code yet.

```bash
$ echo "hello" > test.txt
$ docker run -v `pwd`:/data girder/girder_worker_test:ng reverse /data/test.txt /data/out.txt
Traceback (most recent call last):
  File "./test.py", line 109, in <module>
    cli(obj={})
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "./test.py", line 105, in reverse
    out.write(reversed(f.read()))
TypeError: expected a string or other character buffer object
```

You got to this PR so fast I thought you were a robot. In fact I'm still not sure you aren't 😄

Reply:
@zachmullen 

Haha that's the second time today I've been thought of as a bot! No, I just like looking at all the new python pull requests and am simply trying to learn by exposing myself to a wide variety of projects!

* https://github.com/pulls?page=2&q=is%3Aopen+is%3Apr+language%3Apython&utf8=%E2%9C%93

Reply:
Haha, well thank you for looking out and catching my mistake!

Reply:
No problem! At first I thought it was a new way or something so I just tested out a quick example but I didn't have `click` installed in my current PyCharm project so I wasn't 100% sure.

Reply:
FWIW, to get it working with the `reversed` builtin it would have been:

    ''.join(reversed(f.read()))